### ISSUE TITLE
TestBlockingOwnerRefDoesBlock: panic: duplicate metrics collector registration attempted

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#49974]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/49916/pull-kubernetes-unit/43951/

```
E0801 01:50:48.527040   31465 runtime.go:66] Observed a panic: prometheus.AlreadyRegisteredError{ExistingCollector:(*prometheus.value)(0xc427eec080), NewCollector:(*prometheus.value)(0xc43ba18380)} (duplicate metrics collector registration attempted)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus/registry.go:353
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus/registry.go:152
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/reflector/prometheus/prometheus.go:125
<autogenerated>:8
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector_metrics.go:110
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:106
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:98
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/controller.go:111
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/shared_informer.go:216
/usr/local/go/src/runtime/asm_amd64.s:2197
panic: duplicate metrics collector registration attempted [recovered]
	panic: duplicate metrics collector registration attempted [recovered]
	panic: duplicate metrics collector registration attempted

goroutine 50088 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x126
panic(0x35c01a0, 0xc430c5fb60)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x126
panic(0x35c01a0, 0xc430c5fb60)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc4201dae40, 0xc429f373a0, 0x1, 0x1)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus/registry.go:353 +0x92
k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus.MustRegister(0xc429f373a0, 0x1, 0x1)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus/registry.go:152 +0x53
k8s.io/kubernetes/pkg/util/reflector/prometheus.prometheusMetricsProvider.NewLastResourceVersionMetric(0xc42d038780, 0x53, 0x7f2c018a0620, 0xc4216b0900)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/reflector/prometheus/prometheus.go:125 +0x165
k8s.io/kubernetes/pkg/util/reflector/prometheus.(*prometheusMetricsProvider).NewLastResourceVersionMetric(0x79f2a00, 0xc42d038780, 0x53, 0x7f2c018a0620, 0xc4216b0900)
	<autogenerated>:8 +0x59
k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.newReflectorMetrics(0xc42d038780, 0x53, 0xc42d038780)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector_metrics.go:110 +0x260
k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.NewNamedReflector(0xc459158d70, 0x41, 0x78211c0, 0xc437595a80, 0x39beb40, 0xc459187900, 0x7f2bf55b5990, 0xc4627a3800, 0x0, 0x7f2bf55b5990)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:106 +0x14b
k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.NewReflector(0x78211c0, 0xc437595a80, 0x39beb40, 0xc459187900, 0x7f2bf55b5990, 0xc4627a3800, 0x0, 0xc4369daec0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:98 +0xb0
k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*controller).Run(0xc437677280, 0xc431c5d500)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/controller.go:111 +0x143
k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*sharedIndexInformer).Run(0xc440f88630, 0xc431c5d500)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/shared_informer.go:216 +0x494
created by k8s.io/kubernetes/vendor/k8s.io/client-go/informers.(*sharedInformerFactory).Start
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/informers/factory.go:73 +0x1a0
FAIL	k8s.io/kubernetes/test/integration/garbagecollector	227.852s
```

/cc @caesarxuchao

**Comment by caesarxuchao** [comment]
cc @ironcladlou might be related the periodic sync of GC.

**Comment by ironcladlou** [comment]
https://github.com/kubernetes/kubernetes/pull/49919

**Comment by yujuhong** [comment]
> @caesarxuchao same as the issue fixed by #49919?

Closing based on the comment above
